### PR TITLE
cmd/root: Inspect repo for refs if no commit label

### PR DESCRIPTION
**Pull request by jlebon**
In some automated workflows, it's easier to just write a ref in the
container rather than slap a label on it afterwards (the former can be
done from within the container, while the latter requires the ability
to run code from outside).

Tweak the logic so that we just check for OSTree refs and use the single
ref in there. But always give precedence to the label if present. This
is to address the OpenShift CI use case of creating overlay
oscontainers.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -95,18 +95,6 @@ func pullAndRebase(container string) (imgid string, changed bool) {
 		return
 	}
 
-	// Make sure it has a commit label before pulling
```
Comment (cgwalters) [inline cmd/root.go]: Maybe we could still require some label to be present?  Dunno that it really matters though.

File: cmd/root.go [hunk 2]
```diff
@@ -119,6 +107,30 @@ func pullAndRebase(container string) (imgid string, changed bool) {
...
+		if ostree_version, ok := imagedata.Labels["version"]; ok {
+			glog.Infof("Pivoting to: %s (%s)", ostree_version, ostree_csum)
+		} else {
+			glog.Infof("Pivoting to: %s", ostree_csum)
+		}
+	} else {
+		glog.Infof("No com.coreos.ostree-commit label found in metadata! Inspecting...")
+		refs := strings.Split(utils.RunGetOut("ostree", "refs", "--repo=srv/repo"), "\n")
+		if len(refs) == 1 {
+			glog.Infof("Using ref %s", refs[0])
```
Comment (cgwalters) [inline cmd/root.go]:
```
if refs[0] != "fred" {
  glog.Fatalf("Container appears to be built incorrectly, exiting!")
}
```

**Comment by cgwalters** [comment]
LGTM

File: cmd/root.go (hunk 1 above)
Comment (jlebon) [reply cmd/root.go]: Yeah, I agree. Like a static label that just signals that this is an oscontainer maybe? Then we can error out with a clean error if someone tries to use any other kind of container.

**Pull request merged**